dandischema/datacite/tests/test_datacite.py

```diff
@@ -35,21 +37,178 @@ def datacite_post(datacite: dict, doi: str) -> None:
         auth=("DARTLIB.DANDI", os.environ["DATACITE_DEV_PASSWORD"]),
     )
     rp.raise_for_status()
-
+    print("\n in datacite_post, after posting", doi, rp.status_code)
     # checking if i'm able to get the url
     rg = requests.get(url=f"https://api.test.datacite.org/dois/{doi}/activities")
     rg.raise_for_status()
 
-    # cleaning url
-    _clean_doi(doi)
+    if clean:
+        # cleaning url
+        _clean_doi(doi)
+
+
+def datacite_update(datacite: dict, doi: str) -> None:
+    """Update the datacite object and check the status of the request"""
+    rp = requests.put(
+        url=f"https://api.test.datacite.org/dois/{doi}",
+        json=datacite,
+        headers={"Content-Type": "application/vnd.api+json"},
+        auth=("DARTLIB.DANDI", os.environ["DATACITE_DEV_PASSWORD"]),
+    )
+    rp.raise_for_status()
+
+    # checking if i'm able to get the url
+    rg = requests.get(url=f"https://api.test.datacite.org/dois/{doi}/activities")
+    rg.raise_for_status()
 
 
 def _clean_doi(doi: str) -> None:
     """Remove doi. Status code is ignored"""
-    requests.delete(
+    rq = requests.delete(
         f"https://api.test.datacite.org/dois/{doi}",
         auth=("DARTLIB.DANDI", os.environ["DATACITE_DEV_PASSWORD"]),
     )
+    print("\n in _clean_doi", doi, rq.status_code)
+    return rq.status_code
+
+
+@pytest.mark.skip(
+    reason="to not produced too many dois, not sure if we want to keep it as a test"
```

Comment:
that is because the issue with DOI deletion was not yet addressed on datacite side, right?
then better be mentioned. in the current skip. 

Otherwise -- test should be diligent about removing any DOI it created in its life time.

Reply:
no, the bug on the datacite website is for `draft` doi.

The test creates a `Findable` doi, and this can't be deleted

Reply:
re DOI deletion issue , now we filed
- https://github.com/datacite/bracco/issues/925
- https://github.com/datacite/bracco/issues/924

and indeed, let's not pollute the DOI space there for now (as with regular automated runs).  We might add option to trigger execution once in a while though (monthly?)